packages/server/src/integrations/tests/rest.spec.ts

```diff
@@ -611,4 +633,56 @@ describe("REST Integration", () => {
     expect(calledConfig.headers).toEqual({})
     expect(calledConfig.agent.options.rejectUnauthorized).toBe(false)
   })
+
+  it("uploads file to object store and returns signed URL", async () => {
+    const responseData = Buffer.from("test file content")
+    const filename = "test.tar.gz"
+    const contentType = "application/gzip"
+
+    ;(fetch as unknown as jest.Mock).mockImplementationOnce(() =>
```

Comment:
This is a bit grim, isn't it? Can we not spin up a `minio` container and actually do a proper upload? We spin containers up quite frequently for tests against various databases 🤔

Reply:
Added a test container in so we're using live data. Although this specific part is mocking out the fetch implementation, do you mean grim as in the `unknown` cast?

Reply:
I had originally mistaken this to be related to the objectStore but I realise now it's not. Breaking apart our `node-fetch` mock is a task for another time 😄